### PR TITLE
Improve usage models doc

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @Kludex

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: docs/usage/models.md [hunk 1]
```diff
@@ -272,17 +273,19 @@ class PersonCls:
 
 
 class Pet(BaseModel):
-    model_config = dict(from_attributes=True)
     name: str
     species: str
 
+    model_config = ConfigDict(from_attributes=True)
```
Comment (Kludex) [inline docs/usage/models.md]:
```suggestion
    model_config = ConfigDict(from_attributes=True)

    name: str
    species: str
```

File: docs/usage/models.md [hunk 2]
```diff
@@ -871,19 +860,20 @@ print([pet for pet in pets])
 
 ## Faux immutability
 
-Models can be configured to be immutable via `allow_mutation = False`. When this is set, attempting to change the
+Models can be configured to be immutable via `frozen = True`. When this is set, attempting to change the
```
Comment (Kludex) [inline docs/usage/models.md:863]: Can we add a note about the change from `allow_mutation` to `frozen`?

Comment (hramezani) [reply docs/usage/models.md:893]: added

File: docs/usage/models.md [hunk 3]
```diff
@@ -1111,7 +1079,7 @@ are supported.
 
 ## Parsing data into a specified type
 
-Pydantic includes a standalone utility function `parse_obj_as` that can be used to apply the parsing
+Pydantic includes a standalone utility function `TypeAdapter` that can be used to apply the parsing
```
[the remark this answers is not in the document]
Comment (hramezani) [reply docs/usage/models.md]: I've moved this section to a sub-section under `## Using Pydantic without creating a BaseModel` which `TypeAdapter` introduced first time there

File: docs/usage/models.md [hunk 4]
```diff
@@ -871,19 +890,23 @@ print([pet for pet in pets])
 
 ## Faux immutability
 
-Models can be configured to be immutable via `allow_mutation = False`. When this is set, attempting to change the
+Models can be configured to be immutable via `frozen = True`. When this is set, attempting to change the
 values of instance attributes will raise errors. See [model config](model_config.md) for more details on `Config`.
 
+!!! note
+    Same behavior in Pydantic V1 can be achieved via `allow_mutation = False`. This config is deprecated in Pydantic V2.
```
Comment (tpdorsey) [inline docs/usage/models.md:897]:
```suggestion
    This behavior can be achieved in Pydantic V1 via `allow_mutation = False`. This config is deprecated in Pydantic V2.
```

Comment (tpdorsey) [reply docs/usage/models.md:897]: Does this still communicate the intended meaning? I read this note as indicating what the equivalent behavior was in V1

**Review by Kludex — APPROVED** [review]

**Pull request merged**